submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/api/FilesService.kt

```diff
@@ -1,8 +1,28 @@
 package ac.uk.ebi.biostd.persistence.filesystem.api
 
-import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.NfsFile
+import ebi.ac.uk.io.Permissions
+import java.io.File
 
 internal interface FilesService {
-    fun persistSubmissionFiles(sub: ExtSubmission): ExtSubmission
-    fun cleanSubmissionFiles(sub: ExtSubmission)
+    fun persistSubmissionFile(request: FilePersistenceRequest): ExtFile
+
+    fun cleanSubmissionFile(file: ExtFile)
 }
+
+interface FilePersistenceRequest
```

Comment:
rather than an maker interface I think what you are looking here is a sealed class.